styx-common/src/test/java/com/spotify/styx/util/BasicWorkflowValidatorTest.java

```diff
@@ -88,6 +103,8 @@
   public void setUp() throws Exception {
     MockitoAnnotations.initMocks(this);
     when(dockerImageValidator.validateImageReference(anyString())).thenReturn(Set.of());
+    when(dockerImageValidator.validateImageReference(eq(NOT_VALID_IMAGE)))
```

Comment:
Nothing important, but do we need to mock this? Is that good practice? Or is there another way to do it?

Reply:
Well I guess it depends to whom you ask. 
* Some people are against mocking because they say you should create interface for everything and then implement simple test classes implementation of those interfaces. 
* Other people are against creating "unnecessary" interfaces because that is what mocking libraries are for.  

I see that both are possible approaches but I find Mocking libraries are quite handy and I don't find the downsides of mocking that convincing.

To the question of do we need to mock this? I say that we want verify that we keep calling the validator and react to it as we should, if I comment by mistake the if block that calls the validator then I want a test to fail. So we need a test for that and we want to control when to fail and when not to fail. We could achieve this by mocking (as done in the PR) or we could have introduced a new interface and make the validator implement that interface and create a test class that implements that interface but it is certainly more work that just adding two lines of mocking.

Reply:
Mocking here is ok if we have others tests where we test the behaviour of the `dockerImageValidator.validateImageReference`.
Since this test only makes sure that the method is called, and not that the validation logic behaves properly.
Here are the other tests: https://github.com/spotify/styx/blob/master/styx-common/src/test/java/com/spotify/styx/util/DockerImageValidatorTest.java

Reply:
Ok thank you 👍  Was mostly asking out of curiosity.